# The cert-manager issue continues

looks like https://github.com/jetstack/cert-manager/issues/2498

here's what happened after I pushed a new release adding TLS stuff to kube-prometheus-stack:

```
I0921 20:49:41.102642       1 conditions.go:173] Setting lastTransitionTime for Certificate "alertmanager-hornhorse-cert" condition "Ready" to 2020-09-21 20:49:41.102604463 +0000 UTC m=+38630.669351121
I0921 20:49:41.103715       1 conditions.go:173] Setting lastTransitionTime for Certificate "grafana-hornhorse-cert" condition "Issuing" to 2020-09-21 20:49:41.103697605 +0000 UTC m=+38630.670444300
I0921 20:49:41.105172       1 conditions.go:173] Setting lastTransitionTime for Certificate "grafana-hornhorse-cert" condition "Ready" to 2020-09-21 20:49:41.105152839 +0000 UTC m=+38630.671899521
I0921 20:49:41.106417       1 conditions.go:173] Setting lastTransitionTime for Certificate "prometheus-hornhorse-cert" condition "Ready" to 2020-09-21 20:49:41.106398975 +0000 UTC m=+38630.673145683
I0921 20:49:41.107485       1 conditions.go:173] Setting lastTransitionTime for Certificate "prometheus-hornhorse-cert" condition "Issuing" to 2020-09-21 20:49:41.107468565 +0000 UTC m=+38630.674215260
I0921 20:49:41.111174       1 conditions.go:173] Setting lastTransitionTime for Certificate "alertmanager-hornhorse-cert" condition "Issuing" to 2020-09-21 20:49:41.111151547 +0000 UTC m=+38630.677898254
E0921 20:49:41.564598       1 controller.go:158] cert-manager/controller/CertificateReadiness "msg"="re-queuing item  due to error processing" "error"="Operation cannot be fulfilled on certificates.cert-manager.io \"grafana-hornhorse-cert\": the object has been modified; please apply your changes to the latest version and try again" "key"="prometheus/grafana-hornhorse-cert"
I0921 20:49:41.564957       1 conditions.go:173] Setting lastTransitionTime for Certificate "grafana-hornhorse-cert" condition "Ready" to 2020-09-21 20:49:41.564941013 +0000 UTC m=+38631.131687696
E0921 20:49:41.663444       1 controller.go:158] cert-manager/controller/CertificateReadiness "msg"="re-queuing item  due to error processing" "error"="Operation cannot be fulfilled on certificates.cert-manager.io \"alertmanager-hornhorse-cert\": the object has been modified; please apply your changes to the latest version and try again" "key"="prometheus/alertmanager-hornhorse-cert"
I0921 20:49:41.663846       1 conditions.go:173] Setting lastTransitionTime for Certificate "alertmanager-hornhorse-cert" condition "Ready" to 2020-09-21 20:49:41.663831107 +0000 UTC m=+38631.230577802
E0921 20:49:41.684053       1 controller.go:158] cert-manager/controller/CertificateTrigger "msg"="re-queuing item  due to error processing" "error"="Operation cannot be fulfilled on certificates.cert-manager.io \"prometheus-hornhorse-cert\": the object has been modified; please apply your changes to the latest version and try again" "key"="prometheus/prometheus-hornhorse-cert"
I0921 20:49:41.684554       1 conditions.go:173] Setting lastTransitionTime for Certificate "prometheus-hornhorse-cert" condition "Issuing" to 2020-09-21 20:49:41.684528788 +0000 UTC m=+38631.251275471
E0921 20:49:41.789695       1 controller.go:158] cert-manager/controller/CertificateReadiness "msg"="re-queuing item  due to error processing" "error"="Operation cannot be fulfilled on certificates.cert-manager.io \"prometheus-hornhorse-cert\": the object has been modified; please apply your changes to the latest version and try again" "key"="prometheus/prometheus-hornhorse-cert"
E0921 20:49:42.270529       1 controller.go:158] cert-manager/controller/CertificateKeyManager "msg"="re-queuing item  due to error processing" "error"="Operation cannot be fulfilled on certificates.cert-manager.io \"alertmanager-hornhorse-cert\": the object has been modified; please apply your changes to the latest version and try again" "key"="prometheus/alertmanager-hornhorse-cert"
I0921 20:49:42.610745       1 conditions.go:233] Setting lastTransitionTime for CertificateRequest "alertmanager-hornhorse-cert-nzw9w" condition "Ready" to 2020-09-21 20:49:42.610723213 +0000 UTC m=+38632.177469870
E0921 20:49:42.794159       1 controller.go:158] cert-manager/controller/CertificateKeyManager "msg"="re-queuing item  due to error processing" "error"="Operation cannot be fulfilled on certificates.cert-manager.io \"grafana-hornhorse-cert\": the object has been modified; please apply your changes to the latest version and try again" "key"="prometheus/grafana-hornhorse-cert"
I0921 20:49:44.628994       1 conditions.go:233] Setting lastTransitionTime for CertificateRequest "grafana-hornhorse-cert-v4n8q" condition "Ready" to 2020-09-21 20:49:44.628970615 +0000 UTC m=+38634.195717248
I0921 20:49:47.035011       1 conditions.go:233] Setting lastTransitionTime for CertificateRequest "prometheus-hornhorse-cert-pzrn4" condition "Ready" to 2020-09-21 20:49:47.034984421 +0000 UTC m=+38636.601731041
E0921 20:49:47.949936       1 sync.go:183] cert-manager/controller/challenges "msg"="propagation check failed" "error"="DNS record for \"alertmanager.horn.horse\" not yet propagated" "dnsName"="alertmanager.horn.horse" "resource_kind"="Challenge" "resource_name"="alertmanager-hornhorse-cert-nzw9w-3005501754-3381512916" "resource_namespace"="prometheus" "resource_version"="v1" "type"="DNS-01"
E0921 20:49:47.950429       1 sync.go:183] cert-manager/controller/challenges "msg"="propagation check failed" "error"="DNS record for \"grafana.horn.horse\" not yet propagated" "dnsName"="grafana.horn.horse" "resource_kind"="Challenge" "resource_name"="grafana-hornhorse-cert-v4n8q-2664811883-3703171558" "resource_namespace"="prometheus" "resource_version"="v1" "type"="DNS-01"
E0921 20:49:48.023769       1 sync.go:183] cert-manager/controller/challenges "msg"="propagation check failed" "error"="DNS record for \"alertmanager.horn.horse\" not yet propagated" "dnsName"="alertmanager.horn.horse" "resource_kind"="Challenge" "resource_name"="alertmanager-hornhorse-cert-nzw9w-3005501754-3381512916" "resource_namespace"="prometheus" "resource_version"="v1" "type"="DNS-01"
E0921 20:49:48.058898       1 sync.go:183] cert-manager/controller/challenges "msg"="propagation check failed" "error"="DNS record for \"grafana.horn.horse\" not yet propagated" "dnsName"="grafana.horn.horse" "resource_kind"="Challenge" "resource_name"="grafana-hornhorse-cert-v4n8q-2664811883-3703171558" "resource_namespace"="prometheus" "resource_version"="v1" "type"="DNS-01"
E0921 20:49:49.107658       1 sync.go:183] cert-manager/controller/challenges "msg"="propagation check failed" "error"="DNS record for \"prometheus.horn.horse\" not yet propagated" "dnsName"="prometheus.horn.horse" "resource_kind"="Challenge" "resource_name"="prometheus-hornhorse-cert-pzrn4-1656548385-955803249" "resource_namespace"="prometheus" "resource_version"="v1" "type"="DNS-01"
E0921 20:49:49.198100       1 sync.go:183] cert-manager/controller/challenges "msg"="propagation check failed" "error"="DNS record for \"prometheus.horn.horse\" not yet propagated" "dnsName"="prometheus.horn.horse" "resource_kind"="Challenge" "resource_name"="prometheus-hornhorse-cert-pzrn4-1656548385-955803249" "resource_namespace"="prometheus" "resource_version"="v1" "type"="DNS-01"
I0921 20:51:01.236048       1 acme.go:184] cert-manager/controller/certificaterequests-issuer-acme/sign "msg"="certificate issued" "related_resource_kind"="Order" "related_resource_name"="prometheus-hornhorse-cert-pzrn4-1656548385" "related_resource_namespace"="prometheus" "related_resource_version"="v1" "resource_kind"="CertificateRequest" "resource_name"="prometheus-hornhorse-cert-pzrn4" "resource_namespace"="prometheus" "resource_version"="v1"
I0921 20:51:01.236853       1 conditions.go:222] Found status change for CertificateRequest "prometheus-hornhorse-cert-pzrn4" condition "Ready": "False" -> "True"; setting lastTransitionTime to 2020-09-21 20:51:01.236831912 +0000 UTC m=+38710.803578570
I0921 20:51:01.340319       1 acme.go:184] cert-manager/controller/certificaterequests-issuer-acme/sign "msg"="certificate issued" "related_resource_kind"="Order" "related_resource_name"="grafana-hornhorse-cert-v4n8q-2664811883" "related_resource_namespace"="prometheus" "related_resource_version"="v1" "resource_kind"="CertificateRequest" "resource_name"="grafana-hornhorse-cert-v4n8q" "resource_namespace"="prometheus" "resource_version"="v1"
I0921 20:51:01.341333       1 conditions.go:222] Found status change for CertificateRequest "grafana-hornhorse-cert-v4n8q" condition "Ready": "False" -> "True"; setting lastTransitionTime to 2020-09-21 20:51:01.341314819 +0000 UTC m=+38710.908061501
I0921 20:51:01.342047       1 acme.go:184] cert-manager/controller/certificaterequests-issuer-acme/sign "msg"="certificate issued" "related_resource_kind"="Order" "related_resource_name"="alertmanager-hornhorse-cert-nzw9w-3005501754" "related_resource_namespace"="prometheus" "related_resource_version"="v1" "resource_kind"="CertificateRequest" "resource_name"="alertmanager-hornhorse-cert-nzw9w" "resource_namespace"="prometheus" "resource_version"="v1"
I0921 20:51:01.343750       1 conditions.go:222] Found status change for CertificateRequest "alertmanager-hornhorse-cert-nzw9w" condition "Ready": "False" -> "True"; setting lastTransitionTime to 2020-09-21 20:51:01.34372705 +0000 UTC m=+38710.910473707
E0921 20:51:01.381870       1 sync.go:448] cert-manager/controller/orders "msg"="failed to finalize Order resource due to bad request, marking Order as failed" "error"="403 urn:ietf:params:acme:error:orderNotReady: Order's status (\"valid\") is not acceptable for finalization" "resource_kind"="Order" "resource_name"="alertmanager-hornhorse-cert-nzw9w-3005501754" "resource_namespace"="prometheus" "resource_version"="v1"
E0921 20:51:01.389901       1 sync.go:448] cert-manager/controller/orders "msg"="failed to finalize Order resource due to bad request, marking Order as failed" "error"="403 urn:ietf:params:acme:error:orderNotReady: Order's status (\"valid\") is not acceptable for finalization" "resource_kind"="Order" "resource_name"="grafana-hornhorse-cert-v4n8q-2664811883" "resource_namespace"="prometheus" "resource_version"="v1"
I0921 20:51:01.448173       1 conditions.go:162] Found status change for Certificate "prometheus-hornhorse-cert" condition "Ready": "False" -> "True"; setting lastTransitionTime to 2020-09-21 20:51:01.448148728 +0000 UTC m=+38711.014895336
E0921 20:51:01.512676       1 sync.go:58] cert-manager/controller/orders "msg"="failed to update status" "error"=null "resource_kind"="Order" "resource_name"="alertmanager-hornhorse-cert-nzw9w-3005501754" "resource_namespace"="prometheus" "resource_version"="v1"
E0921 20:51:01.512768       1 controller.go:158] cert-manager/controller/orders "msg"="re-queuing item  due to error processing" "error"="Operation cannot be fulfilled on orders.acme.cert-manager.io \"alertmanager-hornhorse-cert-nzw9w-3005501754\": the object has been modified; please apply your changes to the latest version and try again" "key"="prometheus/alertmanager-hornhorse-cert-nzw9w-3005501754"
I0921 20:51:01.539979       1 conditions.go:162] Found status change for Certificate "grafana-hornhorse-cert" condition "Ready": "False" -> "True"; setting lastTransitionTime to 2020-09-21 20:51:01.539952911 +0000 UTC m=+38711.106699581
I0921 20:51:01.547230       1 conditions.go:162] Found status change for Certificate "alertmanager-hornhorse-cert" condition "Ready": "False" -> "True"; setting lastTransitionTime to 2020-09-21 20:51:01.547206568 +0000 UTC m=+38711.113953225
E0921 20:51:01.623806       1 controller.go:158] cert-manager/controller/CertificateReadiness "msg"="re-queuing item  due to error processing" "error"="Operation cannot be fulfilled on certificates.cert-manager.io \"prometheus-hornhorse-cert\": the object has been modified; please apply your changes to the latest version and try again" "key"="prometheus/prometheus-hornhorse-cert"
I0921 20:51:01.629547       1 conditions.go:162] Found status change for Certificate "prometheus-hornhorse-cert" condition "Ready": "False" -> "True"; setting lastTransitionTime to 2020-09-21 20:51:01.629519495 +0000 UTC m=+38711.196266178
E0921 20:51:02.011229       1 sync.go:58] cert-manager/controller/orders "msg"="failed to update status" "error"=null "resource_kind"="Order" "resource_name"="grafana-hornhorse-cert-v4n8q-2664811883" "resource_namespace"="prometheus" "resource_version"="v1"
E0921 20:51:02.011311       1 controller.go:158] cert-manager/controller/orders "msg"="re-queuing item  due to error processing" "error"="Operation cannot be fulfilled on orders.acme.cert-manager.io \"grafana-hornhorse-cert-v4n8q-2664811883\": the object has been modified; please apply your changes to the latest version and try again" "key"="prometheus/grafana-hornhorse-cert-v4n8q-2664811883"
E0921 20:51:02.012635       1 controller.go:158] cert-manager/controller/CertificateReadiness "msg"="re-queuing item  due to error processing" "error"="Operation cannot be fulfilled on certificates.cert-manager.io \"grafana-hornhorse-cert\": the object has been modified; please apply your changes to the latest version and try again" "key"="prometheus/grafana-hornhorse-cert"
I0921 20:51:02.014637       1 conditions.go:162] Found status change for Certificate "grafana-hornhorse-cert" condition "Ready": "False" -> "True"; setting lastTransitionTime to 2020-09-21 20:51:02.014614805 +0000 UTC m=+38711.581361488
E0921 20:51:02.042811       1 controller.go:196] cert-manager/controller/challenges "msg"="challenge in work queue no longer exists" "error"="challenge.acme.cert-manager.io \"prometheus-hornhorse-cert-pzrn4-1656548385-955803249\" not found"
E0921 20:51:02.218388       1 controller.go:158] cert-manager/controller/CertificateReadiness "msg"="re-queuing item  due to error processing" "error"="Operation cannot be fulfilled on certificates.cert-manager.io \"alertmanager-hornhorse-cert\": the object has been modified; please apply your changes to the latest version and try again" "key"="prometheus/alertmanager-hornhorse-cert"
I0921 20:51:02.219976       1 conditions.go:162] Found status change for Certificate "alertmanager-hornhorse-cert" condition "Ready": "False" -> "True"; setting lastTransitionTime to 2020-09-21 20:51:02.219955202 +0000 UTC m=+38711.786701860
E0921 20:51:02.619425       1 controller.go:158] cert-manager/controller/CertificateReadiness "msg"="re-queuing item  due to error processing" "error"="Operation cannot be fulfilled on certificates.cert-manager.io \"prometheus-hornhorse-cert\": the object has been modified; please apply your changes to the latest version and try again" "key"="prometheus/prometheus-hornhorse-cert"
I0921 20:51:02.624935       1 conditions.go:162] Found status change for Certificate "prometheus-hornhorse-cert" condition "Ready": "False" -> "True"; setting lastTransitionTime to 2020-09-21 20:51:02.624880638 +0000 UTC m=+38712.191627283
I0921 20:51:02.770134       1 request.go:645] Throttling request took 1.113803761s, request: PUT:https://10.96.0.1:443/apis/cert-manager.io/v1/namespaces/prometheus/certificates/grafana-hornhorse-cert/status
E0921 20:51:03.225599       1 controller.go:158] cert-manager/controller/CertificateReadiness "msg"="re-queuing item  due to error processing" "error"="Operation cannot be fulfilled on certificates.cert-manager.io \"grafana-hornhorse-cert\": the object has been modified; please apply your changes to the latest version and try again" "key"="prometheus/grafana-hornhorse-cert"
I0921 20:51:03.232082       1 conditions.go:162] Found status change for Certificate "grafana-hornhorse-cert" condition "Ready": "False" -> "True"; setting lastTransitionTime to 2020-09-21 20:51:03.232056806 +0000 UTC m=+38712.798803451
E0921 20:51:03.433778       1 controller.go:158] cert-manager/controller/CertificateReadiness "msg"="re-queuing item  due to error processing" "error"="Operation cannot be fulfilled on certificates.cert-manager.io \"alertmanager-hornhorse-cert\": the object has been modified; please apply your changes to the latest version and try again" "key"="prometheus/alertmanager-hornhorse-cert"
I0921 20:51:03.436803       1 conditions.go:162] Found status change for Certificate "alertmanager-hornhorse-cert" condition "Ready": "False" -> "True"; setting lastTransitionTime to 2020-09-21 20:51:03.436769485 +0000 UTC m=+38713.003516280
E0921 20:51:03.866580       1 controller.go:196] cert-manager/controller/challenges "msg"="challenge in work queue no longer exists" "error"="challenge.acme.cert-manager.io \"alertmanager-hornhorse-cert-nzw9w-3005501754-3381512916\" not found"
E0921 20:51:04.054883       1 controller.go:196] cert-manager/controller/challenges "msg"="challenge in work queue no longer exists" "error"="challenge.acme.cert-manager.io \"grafana-hornhorse-cert-v4n8q-2664811883-3703171558\" not found"
W0921 20:54:30.261912       1 warnings.go:67] extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
```

Well, I took about 15 minutes to investigate this issue, and in that time it fixed itself, so :confused:
